### DISCUSSION TITLE
请问如何启用Peer 白名单/黑名单 (How to enable Peer whitelist/blacklist)

### DISCUSSION THREAD

**Discussion by Radium-bit** [c0re100/qBittorrent-Enhanced-Edition#528] [Q&A]
由于发现某一个Peer经常吸血，因此我想将这个Peer标识添加到黑名单中，请问如何操作？
我在Readme中没有找到对应指引。
对于这个Peer标识已经有部分讨论，在[彗星论坛](https://www.cometbbs.com/t/%E6%83%B3%E9%97%AE%E9%97%AE%E5%A4%A7%E5%AE%B6%E6%80%8E%E4%B9%88%E7%9C%8B%E8%BF%99%E7%B1%BB%E7%94%A8%E6%88%B7%E7%9A%84%E5%BC%82%E5%B8%B8%E6%83%85%E5%86%B5/90636)和[anacrolix/torrent](https://github.com/anacrolix/torrent/issues/889)的Issue中都有所提及。

**Comment by ZetaX-XD** [comment]
首先是我看到过的两个实例
- `-GT\d{4}-    .*` 
 屏蔽所有Peer ID客户端为 `GT\d{4}` 的客户端（\d{4}意思是任意四个数字，.*意思是任意字符串）

- `-GT0003- github.com/anacrolix/torrent\s\(devel\)\s\(anacrolix/torrent\sunknown\)` 
 屏蔽Peer ID客户端为`GT1003`，且客户端为`github.com/anacrolix/torrent (devel) (anacrolix/torrent unknown)`的用户

以下是我个人的、不全面的理解，
先在 qbittorrent 的目录下，创建一个`Data`文件夹，
然后在这个文件夹中，创建`peer_blacklist.txt`文件，用正则规则来表示想要屏蔽的内容
参考自 #251

然而，我按这样设置完全没有作用

**Reply by JZYFC** [reply]
关于peer_blacklist.txt等文件的存放位置

* Windows: `%LocalAppData%\qBittorrent`
* Linux: `$HOME/.local/share/qBittorrent`

启动qBittorrent EE后，可在日志（菜单栏中：视图 -> 日志 -> 显示）中看到如下内容

```
<时间戳> - 'peer_whitelist.txt' doesn't exist. The corresponding filter is disabled.
<时间戳> - 'peer_blacklist.txt' contains 2 valid rules.
```

 参考自：https://github.com/c0re100/qBittorrent-Enhanced-Edition/pull/251
